FreeScape/GameObjects/Player/Player.cs

```diff
@@ -5,42 +5,40 @@
 using FreeScape.Engine.Managers;
 using FreeScape.Engine.Physics;
 using FreeScape.Engine.Physics.Movements;
-using FreeScape.Engine.Physics.Collisions;
 using FreeScape.Engine.Physics.Collisions.Colliders;
 using FreeScape.Engine.Providers;
+using FreeScape.Engine.Render.Animations;
 using SFML.Graphics;
 
 namespace FreeScape.GameObjects
 {
-    public class Player : PlayerActions, IMovable, ICollidable 
+    public class Player : IMovable, ICollidable 
     {
         private readonly UserInputMovement _movement;
         private readonly DisplayManager _displayManager;
+        private readonly AnimationProvider _animationProvider;
         private CircleShape _shape;
         public int ZIndex { get; set; }
         public float Weight { get; set; }
         public Vector2 Size { get; set; }
         public Vector2 Scale { get; set; }
-        private float speed = 0;
-        public float Speed { 
-            get { return speed * PlayerActionSpeedModifier; }
-            set { speed = value; } }
+        public float Speed { get; set; }
         public HeadingVector HeadingVector { get; private set; }
         public List<ICollider> Colliders { get; set; }
-        public Vector2 Velocity { get; set; }
         public Vector2 Position { get; set; }
+        private DirectionAnimation _currentAnimation;
```

Comment:
Probably need to work on an animation stack so we can go back to previous animations like.... walking -> Sword swing -> walking

Reply:
I'll create a issue for this

Reply:
#20